ui/selectors/transactions.js

```diff
@@ -271,47 +277,160 @@ export const nonceSortedTransactionsSelector = createSelector(
+          !currentTransaction.isEphemeral;
+
+        // If a transaction failed on chain or was confirmed then it should
+        // always be the primary because no other transaction is more valid.
+        currentTransaction.shouldBePrimary =
+          currentTransaction.isConfirmed || currentTransaction.isOnChainFailure;
+
+        // Primary Transaction Logic Cases
+        // --------------------------------------------------------------------
+        // Primary transaction: The transaction for any given nonce which has
+        // the most valid status on the network.
+        // Example:
+        // 1. Submit transaction A
+        // 2. Speed up Transaction A.
+        // 3. This creates a new Transaction (B) with higher gas params.
+        // 4. Transaction A and Transaction B are both submitted.
+        // 5. We expect Transaction B to be the most valid transaction to use
+        //    for the status of the transaction group because it has higher
+        //    gas params and should be included first.
+        // The following logic variables are used for edge cases that protect
+        // against UI bugs when this breaks down.
+        const previousPrimaryTransaction = {
+          // As we loop through the transactions in state we may temporarily
+          // assign a primaryTransaction that is an "Ephemeral" transaction,
+          // which is one that failed before being broadcast for inclusion in a
+          // block. When this happens, and we have another transaction to
+          // consider in a nonce group, we should use the new transaction.
+          isEphemeral:
+            nonceProps.primaryTransaction.status ===
+              TRANSACTION_STATUSES.FAILED &&
+            nonceProps.primaryTransaction?.txReceipt?.status !== '0x0',
+        };
+
+        // Initial Transaction Logic Cases
+        // --------------------------------------------------------------------
+        // Initial Transaction: The transaciton that most likely represents the
+        // user's intent when creating/approving the transaction. In most cases
+        // this is the first transaction of a nonce group, by time, but this
+        // breaks down in the case of users with the advanced setting enabled
+        // to set their own nonces manually. In that case a user may submit two
+        // completely different transactions of the same nonce and they will be
+        // bundled together by this selector as the same activity entry.
+        const previousInitialTransaction = {
+          // As we loop through the transactions in state we may temporarily
+          // assign a initialTransaction that is an "Ephemeral" transaction,
+          // which is one that failed before being broadcast for inclusion in a
+          // block. When this happens, and we have another transaction to
+          // consider in a nonce group, we should use the new transaction.
+          isEphemeral:
             TRANSACTION_STATUSES.FAILED &&
```

Comment:
@brad-decker Looks like line 403 needs fixing.

Should it be `nonceProps.initialTransaction.status === TRANSACTION_STATUSES.FAILED &&`?

Reply:
WOOPS good catch!

Reply:
@danjm done in latest.